tests/constructors/inspection_test.py

```diff
@@ -41,6 +41,16 @@ def test_new_type_underlying_type_retrieved():
     assert product_spec.returned_type is int
 
 
+def test_multi_inheritied_new_type_underlying_type_retrieved():
+    from typing import NewType
+
+    new_type = NewType("new_type", int)
+    new_new_type = NewType("new_new_type", new_type)
```

Comment:
Is this actually used in practice?

Reply:
Yes...! I accidentally made a `NewType` of a `NewType` while writing multiple workflows...
And there is no reason not to support these I thought.